### PR TITLE
Add some fedmsg-publishing karma commands.

### PR DESCRIPTION
**Pull request by ralphbean**
This adds USERNAME++ commands to zodbot.  It keeps track of who gave who what
so that people can't abuse it, and it publishes fedmsgs when karma is awarded.
We'll use that to trigger badges.

### CODE REVIEW

**Comment by ralphbean** [comment]
The bot is running right now in ``#threebean`` if you want to test it out.

**Comment by ralphbean** [comment]
The artwork for these badges is already done (which is what prompted me to get this written).  All here:  https://fedorahosted.org/fedora-badges/ticket/328

File: config.py [hunk 1]
```diff
@@ -65,4 +65,17 @@ def configure(advanced):
                     private=True))
 
 
+conf.registerGroup(Fedora, 'karma')
+conf.registerGlobalValue(
+    Fedora.karma, 'db_path',
+    registry.String('/var/tmp/supybot-karma.db',
+                    """Path to a karma db on disk"""))
```
Comment (pypingou) [inline config.py]: supybot provides an interface to connect to the DB directly?
I'm not seeing any DB related code in this p-r.

File: config.py [hunk 2]
```diff
@@ -65,4 +65,17 @@ def configure(advanced):
...
+conf.registerGlobalValue(
+    Fedora.karma, 'unaddressed',
+    registry.Boolean(True, "Allow unaddressed karma commands"))
```
Comment (pypingou) [inline config.py]: I am not quite sure to understand what this option means, could we expand on it somewhere?

File: plugin.py [hunk 3]
```diff
@@ -630,6 +642,151 @@ def swedish(self, irc, msg, args):
...
+                self._do_karma(irc, channel, agent, recip, explicit=False)
+
+
+    def karma(self, irc, msg, args, name):
+        """<username>
+
+        Return the total karma for a FAS user."""
+        data = None
+        try:
+            data = shelve.open(self.karma_db_path)
```
Comment (ralphbean) [inline plugin.py]: @pypingou, the db code is right here.

**Pull request merged**